test/externalTests/runners/foundry.py

```diff
@@ -0,0 +1,182 @@
+        check=True
+    )
+
+
+class FoundryRunner(TestRunner):
+    """Configure and run Foundry-based projects"""
+    foundry_config_file = "foundry.toml"
+
+    def __init__(self, config: TestConfig, setup_fn=None, compile_fn=None, test_fn=None):
+        super().__init__(config)
+        self.setup_fn = setup_fn
+        self.compile_fn = compile_fn
+        self.test_fn = test_fn
+        self.env = os.environ.copy()
+        # Note: the test_dir will be set on setup_environment
+        self.test_dir: Path
+
+    def setup_environment(self, test_dir: Path):
+        """Configure the project build environment"""
+
+        print("Configuring Foundry building environment...")
+        self.test_dir = test_dir
+        if which("forge") is None:
+            raise RuntimeError("Forge not found.")
+        if self.setup_fn:
+            self.setup_fn(self.test_dir, self.env)
+
+    @staticmethod
+    def profile_name(preset: str):
+        """Returns foundry profile name"""
+        # Replace - or + by underscore to avoid invalid toml syntax
+        return re.sub(r"(\-|\+)+", "_", preset)
+
+    @staticmethod
+    def profile_section(profile_fields: dict) -> str:
+        return dedent(
+            """\
+            [profile.{name}]
+            gas_reports = ["*"]
+            auto_detect_solc = false
+            solc = "{solc}"
+            evm_version = "{evm_version}"
+            optimizer = {optimizer}
+            via_ir = {via_ir}
+
+            [profile.{name}.optimizer_details]
+            yul = {yul}
+            """
+        ).format(
+            name=profile_fields["name"],
+            solc=profile_fields["solc"],
+            evm_version=profile_fields["evm_version"],
+            optimizer=profile_fields["optimizer"],
+            via_ir=profile_fields["via_ir"],
+            yul=profile_fields["yul"]
+        )
+
+    @TestRunner.on_local_test_dir
+    def clean(self):
+        """Clean the build artifacts and cache directories"""
+        run_forge_command("forge clean")
+
+    @TestRunner.on_local_test_dir
+    def compiler_settings(self, solc_version: str, presets: List[str]):
+        """Configure forge tests profiles"""
+
+        binary_type = self.config.solc.binary_type
+        binary_path = self.config.solc.binary_path
+        print(dedent(
+            f"""\
+            Configuring Forge profiles...
+            -------------------------------------
+            Config file: {self.foundry_config_file}
+            Binary type: {binary_type}
+            Compiler path: {binary_path}
+            -------------------------------------
+            """
+        ))
+
+        # TODO: Add support to solcjs. Currently only native solc is supported. # pylint: disable=fixme
+        if binary_type == "solcjs":
+            raise NotImplementedError(
+                "Solcjs binaries are currently not supported with Foundry. Please use `native` binary_type."
+            )
```

Comment:
I kept the `binary_type` though. Since it is passed by the CI to the script, so it still compatible with the old external tests until we migrate everything to python. But I moved the solcjs and soljson.js part to another branch.